components/Portfolio/AssetDisplay/AssetDisplay.tsx

```diff
@@ -1,36 +1,221 @@
-import type { Asset } from "@/utils/types";
+import type {
+  Asset,
+  AssetCurrent,
+  AssetHistory,
+} from "@/utils/types";
 
-import {
-  useAssetCurrentQueries,
-  useAssetHistoryQueries,
-} from "@/hooks/useAssetQueries";
+import { assetDisplayData } from "@/utils/assetDisplayData";
+import { cn } from "@/utils/cn";
+import { currencyMap } from "@/utils/maps";
 import { extractDate } from "@/utils/extractDate";
+import { localeFormat } from "@/utils/formatHelpers";
+import { useId } from "react";
+
+import CaretIcon from "@/Icons/Caret";
+import Image from "next/image";
+import { Infinity as InfinityIcon } from "lucide-react";
+import ProgressWidget from "@/components/ProgressWidget";
+import { SquarePen as SquarePenIcon } from "lucide-react";
+import { useUserCurrencySetting } from "@/hooks/useUserSettings";
 
 type Props = {
   asset: Asset;
-  currentResponse: ReturnType<typeof useAssetCurrentQueries>[number];
-  historyResponse: ReturnType<typeof useAssetHistoryQueries>[number];
+  assetCurrent: AssetCurrent | undefined;
+  assetHistory: AssetHistory | undefined;
 };
 
-const AssetDisplay = ({ asset, currentResponse, historyResponse }: Props) => {
-  const { coinName, date } = asset;
+const AssetDisplay = ({ asset, assetCurrent, assetHistory }: Props) => {
+  const { coinName, coinSymbol, coinImage, date } = asset;
+  const currency = useUserCurrencySetting();
+
+  const editButtonId = useId();
+  const displayDate = extractDate(date).toLocaleString("en-US", {
+    dateStyle: "medium",
+  });
+
+  const maybeDisplayData = assetDisplayData(
+    asset,
+    currency,
+    assetCurrent,
+    assetHistory
+  );
+
+  const placeholder = <span className="animate-pulse">--</span>;
+
+  const currentValue = (() => {
```

Comment:
Could be simplified to => 

const currentValue = maybeDisplayData?.currentValue
  ? <span>{currencyMap.get(currency) + localeFormat(maybeDisplayData.currentValue)}</span>
  : placeholder;